app/views/building_map/index.html.erb

```diff
@@ -1,11 +1,141 @@
-<%=
-  map(
+<div class="accordion mb-2 accordion-flush" id="routingInput">
+	<div class="accordion-item">
+		<h2 class="accordion-header" id="headingOne">
+			<button class="accordion-button collapsed" type="button" data-bs-toggle="collapse"
+				data-bs-target="#collapseOne" aria-expanded="true" aria-controls="collapseOne">
+				<i class="fas fa-directions directions-icon"></i>
+			</button>
+		</h2>
+		<div id="collapseOne" class="accordion-collapse collapse" aria-labelledby="headingOne"
+			data-bs-parent="#routingInput">
+			<div class="accordion-body">
+				<form class="row gy-2 gx-3 align-items-center" autocomplete="off" id="navigation_form">
+					<div class="col-auto">
+						<div class="input-group">
+							<div class="input-group-text"><i class="fas fa-map-marker-alt route-start-icon"></i></div>
+							<input name='start' type="text" class="form-control" list="startOptions" id="start_input"
+								placeholder="From" onchange="request_location()">
+						</div>
+					</div>
+					<div class="col-auto">
+						<div class="input-group">
+							<div class="input-group-text"><i class="fas fa-map-marker-alt route-end-icon"></i></div>
+							<input name='dest' type="text" class="form-control" list="destOptions" id="dest_input"
+								placeholder="To">
+						</div>
+					</div>
+					<div class="col-auto">
+						<button type="submit" formaction="/building_map" class="btn btn-primary" onclick="request_location()">Go</button>
+					</div>
+				</form>
+			</div>
+		</div>
+	</div>
+</div>
+
+<datalist id="startOptions">
+	<option>Your location</option>
+	<%= Places::DESTINATIONS.each do |dest, coords| %>
+		<option>
+			<%= dest %>
+		</option>
+		<% end %>
+</datalist>
+<datalist id="destOptions">
+	<%= Places::DESTINATIONS.each do |dest, coords| %>
+		<option>
+			<%= dest %>
+		</option>
+		<% end %>
+</datalist>
+
+<%= map(
     center: {
-        # HPI coordinates if no center is set
-        latlng: !@target.nil? ? @target.split(",") : [52.39339, 13.13208],
-        zoom: 18,
+        # HPI coordinates
+        latlng: RoutingHelper.center(@start),
```

Comment:
To be consistent with what we do for polygons, polylines, etc: This should probably live in the `BuildingMapHelper` module and be called `leaflet_`-something (and possibly return the entire object, including the zoom)